test/sys/test_pthread.rs

```diff
@@ -20,3 +20,103 @@ fn test_pthread_kill_none() {
     pthread_kill(pthread_self(), None)
         .expect("Should be able to send signal to my thread.");
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_pthread_mutex_wrapper() {
+    use nix::{
+        sys::{
+            mman::{mmap, MapFlags, ProtFlags},
+            pthread::{Mutex, MutexAttr},
+        },
+        unistd::{fork, ForkResult},
+    };
+    use std::{mem::size_of, num::NonZeroUsize, os::unix::io::OwnedFd};
+    struct MutexWrapper {
+        lock: Mutex,
+        data: u128,
+    }
+    impl MutexWrapper {
+        fn add(&mut self) {
```

Comment:
There's little point to using a mutex if one already has a `&mut self` reference.  So this isn't a very realistic test.

Reply:
I've changed this to `&self`.